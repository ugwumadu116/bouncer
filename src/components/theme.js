import * as colors from "./color/color";
import * as fontsize from "./text/fontsize";
import * as fontweight from "./text/fontweight";
import * as lineheight from "./text/lineheight";
import * as letterspacing from "./text/letterspacing";
import * as btnSizes from "./button/button";
import * as btnState from "./button/btnState";
import * as inputsizes from './form/input/inputsizes'
import * as inputbuttonSize from './form/inputButton/inputbuttonsizes'
import * as selectSize from './form/select/selectSizes'


const theme = {
  colors,
  fontsize,
  fontweight,
  lineheight,
  letterspacing,
  btnSizes,
  btnState,
  inputsizes,
  inputbuttonSize,
  selectSize
};
export default theme;
