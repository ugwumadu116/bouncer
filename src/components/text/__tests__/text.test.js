import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'
import Text from "../"
import theme from '../../theme'


it('renders correctly where only font size is set', ()=>{
    const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Text small >hello</Text>  
    </ThemeProvider>
    
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('renders correctly where letter spacing is set',()=>{
   
    const tree = renderer.create(
    <ThemeProvider theme={theme}>
    <Text ltSpacing1>Hello</Text>
    </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot()
})

it('renders correctly where font weight is set',()=>{
   
    const tree = renderer.create(
    <ThemeProvider theme={theme}>
    <Text bold>Hello</Text>
    </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot()
})

it('renders correctly where fline height is set',()=>{
   
    const tree = renderer.create(
    <ThemeProvider theme={theme}>
    <Text ht5>Hello</Text>
    </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot()
})



