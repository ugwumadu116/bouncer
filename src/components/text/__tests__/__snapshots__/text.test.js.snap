// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly where fline height is set 1`] = `
<p
  className="sc-bdVaJa fiqdtG"
>
  Hello
</p>
`;

exports[`renders correctly where font weight is set 1`] = `
<p
  className="sc-bdVaJa ejaqKF"
>
  Hello
</p>
`;

exports[`renders correctly where letter spacing is set 1`] = `
<p
  className="sc-bdVaJa hbjBEG"
>
  Hello
</p>
`;

exports[`renders correctly where only font size is set 1`] = `
<p
  className="sc-bdVaJa GZDkD"
>
  hello
</p>
`;
